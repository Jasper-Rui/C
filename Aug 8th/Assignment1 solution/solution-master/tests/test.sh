#!/bin/sh

echo ======================================================
echo ====================== TEST $1 ========================
echo ======================================================
if timeout 10 ./miner/miner < tests/test.$1.in > tests/test.$1.out; then 
  if diff tests/test.$1.out tests/test.$1.expected > /dev/null; then
    echo PASSED
  else
    echo FAILED
    echo ======
    echo Expected Output:
    echo ++++++++++++++++
    cat tests/test.$1.expected
    echo =====================
    echo Actual Output:
    echo ++++++++++++++++
    cat tests/test.$1.out
    exit 1
  fi
elif [ $? == 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: the program crashed
  echo Exit code $?
  exit 1
fi
