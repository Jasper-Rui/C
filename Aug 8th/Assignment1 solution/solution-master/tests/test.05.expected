Block mined: 1 0 0x00000000 0
0x00000060 0x00a71277
Block mined: 2 1 0x00a71277 0
0x00000035 0x006a8f24
